iquip/apps/explorer.py

```diff
@@ -12,82 +12,88 @@
 
 from iquip import cmdtools
 
-class ExplorerFrame(QWidget):  # pylint: disable=too-few-public-methods
+class ExplorerFrame(QWidget):
     """Frame for showing the experiment list and opening an experiment."""
 
     def __init__(self, parent: Optional[QWidget] = None):
         """Extended."""
         super().__init__(parent=parent)
         # widgets
-        self.expStructure = QTreeWidget(self)
-        self.expStructure.header().setVisible(False)
+        self.fileTree = QTreeWidget(self)
+        self.fileTree.header().setVisible(False)
         self.reloadButton = QPushButton("Reload", self)
         self.openButton = QPushButton("Open", self)
         # layout
         layout = QVBoxLayout(self)
         layout.addWidget(self.reloadButton)
-        layout.addWidget(self.expStructure)
+        layout.addWidget(self.fileTree)
         layout.addWidget(self.openButton)
 
 
 class ExplorerApp(qiwis.BaseApp):  # pylint: disable=too-few-public-methods
     """App for showing the experiment list and opening an experiment."""
 
     def __init__(self, name: str, masterPath: str = ".", parent: Optional[QObject] = None):
-        """Extended."""
+        """Extended.
+
+        Args:
+            masterPath: The path where artiq_master command is running.
+        """
         super().__init__(name, parent=parent)
-        self.repositoryPath = os.path.join(masterPath, "repository")
+        self.repositoryPath = posixpath.join(masterPath, "repository")
         self.explorerFrame = ExplorerFrame()
-        self.loadExpStructure()
+        self.loadFileTree()
         # connect signals to slots
-        self.explorerFrame.reloadButton.clicked.connect(self.loadExpStructure)
-        self.explorerFrame.openButton.clicked.connect(self.openExp)
+        self.explorerFrame.reloadButton.clicked.connect(self.loadFileTree)
+        self.explorerFrame.openButton.clicked.connect(self.openExperiment)
 
     @pyqtSlot()
-    def loadExpStructure(self):
-        """Loads the experiment file structure in self.explorerFrame.expStructure.
+    def loadFileTree(self):
+        """Loads the experiment file structure in self.explorerFrame.fileTree.
 
         It assumes that all experiment files are in self.repositoryPath.
         """
-        self.explorerFrame.expStructure.clear()
-        self._addExpFile(self.repositoryPath, self.explorerFrame.expStructure)
+        self.explorerFrame.fileTree.clear()
+        self._addFile(self.repositoryPath, self.explorerFrame.fileTree)
+
+    def _addFile(self, path: str, parent: Union[QTreeWidget, QTreeWidgetItem]):
+        """Searches all files in path and adds them into parent.
 
-    def _addExpFile(self, path: str, parent: Union[QTreeWidget, QTreeWidgetItem]):
-        """Searches the sub elements and add them into self.explorerFrame.expStructure.
+        A directory which starts with "_" will be ignored, e.g. __pycache__/.
```

Comment:
Now it includes files that are not directories!

Reply:
Oh, I will modify it!